lib/dai-plugin-mcd/src/CdpManager.js

```diff
@@ -0,0 +1,94 @@
+import Maker from '@makerdao/dai';
+import { ServiceRoles } from './constants';
+import assert from 'assert';
+import ManagedCdp from './ManagedCdp';
+import { getIlkForCurrency } from './utils';
+import { padStart } from 'lodash';
+const { ETH } = Maker;
+
+export default class CdpManager extends Maker.LocalService {
+  constructor(name = ServiceRoles.CDP_MANAGER) {
+    super(name, ['smartContract', 'accounts', 'proxy']);
+  }
+
+  async open() {
+    assert(this.get('proxy').currentProxy(), 'User has no DSProxy');
```

Comment:
In #70 I added an error to the proxy service's execute function that throws if the user doesn't have a proxy, so this assertion can probably be removed